src/subdomains/supporting/address-pool/route/deposit-route.service.ts

```diff
@@ -0,0 +1,111 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { merge } from 'lodash';
+import { Blockchain } from 'src/integration/blockchain/shared/enums/blockchain.enum';
+import { DfxLogger } from 'src/shared/services/dfx-logger';
+import { PaymentLink } from 'src/subdomains/core/payment-link/entities/payment-link.entity';
+import { FindOneOptions, In } from 'typeorm';
+import { DepositRoute } from './deposit-route.entity';
+import { DepositRouteRepository } from './deposit-route.repository';
+
+@Injectable()
+export class DepositRouteService {
+  private readonly logger = new DfxLogger(DepositRouteService);
+
+  constructor(private readonly depositRouteRepo: DepositRouteRepository) {}
+
+  async get(userId: number, id: number): Promise<DepositRoute> {
+    const routes = await this.depositRouteRepo.findOne({
+      where: { id, user: { id: userId } },
+      relations: { user: { userData: true } },
+    });
+    if (!routes) throw new NotFoundException('Route not found');
+    return routes;
+  }
+
+  async getById(id: number, options?: FindOneOptions<DepositRoute>): Promise<DepositRoute> {
+    const defaultOptions = { where: { id }, relations: { user: { userData: true } } };
+    return this.depositRouteRepo.findOne(merge(defaultOptions, options));
+  }
+
+  async getLatest(userId: number): Promise<DepositRoute | null> {
+    return this.depositRouteRepo.findOne({
+      where: { user: { id: userId } },
+      relations: { user: { userData: true } },
+      order: { created: 'DESC' },
+    });
+  }
+
+  async getByLabel(userId: number, label: string, options?: FindOneOptions<DepositRoute>): Promise<DepositRoute> {
+    const defaultOptions = {
+      where: { route: { label }, user: { id: userId } },
+      relations: { user: { userData: true } },
+    };
+    return this.depositRouteRepo.findOne(merge(defaultOptions, options));
+  }
+
+  validateLightningRoute(route: DepositRoute): void {
+    if (!route) throw new NotFoundException('Route not found');
+    if (route.deposit.blockchains !== Blockchain.LIGHTNING)
+      throw new BadRequestException('Only Lightning routes are allowed');
+  }
```

Comment:
Do we really need that method? It is only used on 2 other locations? For me its a bit strange with only 2 if statements in it 🤔